packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
 
-  /// widget builder for creating cards
-  final CardsBuilder cardsBuilder;
+  /// The indexed widget builder that builds a card for the given index.
+  final IndexedWidgetBuilder cardBuilder;
 
-  ///cards count
-  final int cardsCount;
+  /// The number of cards in the stack.
+  final int cardCount;
 
-  ///cards spacing
+  /// Cards spacing.
   final double cardsSpacing;
 
-  ///background cards count
-  final int backgroundCardsCount;
-
-  /// controller to trigger unswipe action
+  /// Background cards count
+  final int backgroundCardCount;
+
+  /// The amount to scale each successive background card down by,
+  ///
+  /// The difference in scale for each background card is a fixed amount
+  /// relative to the original card.
+  ///
+  /// Defaults to .9.
+  final double backgroundCardScale;
+
+  /// The amount to offset each successive background card relative to the card
+  /// before it.
+  ///
+  /// Defaults to offsetting each card down by 40 dp.
+  final Offset? backgroundCardOffset;
+
+  /// A controller that provides programmatic control of the swiper and notifies
+  /// on swiper state changes.
   final AppinioSwiperController? controller;
 
-  /// duration of every animation
+  /// The duration of swipe animations.
+  ///
+  /// Swipe animations start after the user lifts their finger or when a drag
+  /// is triggered by [AppinioSwiperController].
   final Duration duration;
 
-  /// add listener to check when the card is swiping
-  final void Function(AppinioSwiperDirection direction)? onSwiping;
+  /// A callback that is called whenever the card moves.
+  final void Function(AxisDirection direction, Offset position)?
+      onCardPositionChanged;
 
-  /// padding of the swiper
-  final EdgeInsetsGeometry padding;
-
-  /// maximum angle the card reaches while swiping
+  /// The maximum angle the card reaches while horizontally swiping.
+  ///
+  /// Cards lean in the direction of the swipe to sell their physicality. Set
+  /// this to 0 to disable the lean.
   final double maxAngle;
 
-  /// set to true if verticalSwipe should be disabled, exception: triggered from the outside
-  final AppinioSwipeOptions swipeOptions;
-
-  /// threshold from which the card is swiped away
-  final int threshold;
-
-  /// set to true if swiping should be disabled, exception: triggered from the outside
+  /// Sets whether the card should angle in the opposite direction when it is
+  /// dragged from the bottom half.
+  ///
+  /// Defaults to true.
+  final bool invertAngleOnBottomDrag;
+
+  /// What swipe directions to allow.
+  ///
+  /// Swipes triggered by a controller are always allowed and are not affected
+  /// by [swipeOptions].
+  final SwipeOptions swipeOptions;
+
+  /// The minimum distance a user has to pan the card before triggering a swipe
+  /// animation.
+  ///
+  /// If a pan is less than [threshold] the card will animate back to it's
+  /// original position and the card stack will not change.
+  final double threshold;
```

Comment:
This should really be replaced with a ballistic simulation that swipes on fling regardless of distance dragged.
We could rip off the snap behavior I wrote for `DraggableScrollableSheet`:
https://github.com/flutter/flutter/blob/e12d1a798c4984c81de284bdef9ae6687624725f/packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart#L1127